crypto/src/lib.rs

```diff
@@ -210,14 +210,52 @@ impl KeyPair {
         self.private_key.digest_function()
     }
 
-    /// Construct `KeyPair`
-    pub fn new(public_key: PublicKey, private_key: PrivateKey) -> Self {
+    /// Construct `KeyPair` from a matching pair of public and private key.
+    /// It is up to the user to ensure that the given keys indeed make a pair.
+    #[cfg(not(feature = "std"))]
+    pub fn new_unchecked(public_key: PublicKey, private_key: PrivateKey) -> Self {
```

Comment:
I'd like to make it `unsafe` as `std` does for it's unchecked funtions (i.e. [`String::from_utf8_unchecked()`](https://doc.rust-lang.org/std/string/struct.String.html#method.from_utf8_unchecked))

Reply:
I'd like that too, but that would be a misuse of the `unsafe`. `unsafe` means a possible violation of memory access. Either `unchecked` suffix doesn't imply unsafe in Rust std and it's good as it is, or I can rename the method to something else